[develop] Consolidate verification tasks using retrieve_data.py

mkavulich opened the pull request:
## DESCRIPTION OF CHANGES: 
As described in Issue #850, the scripts for retrieving observation data for verification are quite convoluted and hard-to-maintain due to the constant changing of naming conventions on HPSS. We can leverage the functionality of the `retrieve_data.py` script to greatly simplify and consolidate this code, and that's what this PR achieves.

In addition, many other bug fixes and improvements are included:

 - The j-Jobs and exscripts for the retrieving of CCPA, MRMS, NDAS, and NOHRSC data are consolidated into a single set of scripts.
 - Verification tests that use staged data on disk no longer need to manually deactivate "get_obs" tasks; now tasks check if staged data was specified and ensure it is all present
 - Obs locations (CCPA_OBS_DIR, MRMS_OBS_DIR, NDAS_OBS_DIR, and NOHRSC_OBS_DIR) no longer need a specific directory structure, though the file format remains the same
 - Major improvements to `mrms_pull_topofhour.py`; now has a `main()` function for importing, uses proper command-line arguments with argparse, uses f-strings to simplify print statements, and raises an exception if no observation file is found instead of failing silently
 - Fixes bug in `retrieve_data.py` where filenames in tarballs containing full paths are not properly extracted
 - Removes "choices" for `--data_type` argument in `retrieve_data.py`: the data type is customizable via the config file (which is also customizable), so we shouldn't hard-code arbitrary choices. Missing data types are gracefully handled later anyway.
 - Add all verification tests to a coverage suite
 - Add new test `custom_ESGgrid_Great_Lakes_snow_8km` that specifies a custom domain for a snowstorm in February 2023, re-using data that was staged for the AQM test. This is part of an effort to expand testing (finishing Issue #587) that will be completed in a subsequent PR, including explicit tests for winter precipitation verification.
 - Add cold-air-damming test to all comprehensive test lists, not just orion.
 - ~~Fixes bug described in Issue #851.~~ Test `MET_ensemble_verification_only_vx_time_lag` now works on most platforms, not just hard-coded to Hera
 - Comments related to verification obs in config_defaults.yaml are consolidated and relevant information moved to new consolidated obs-pulling script

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## TESTS CONDUCTED: 
Ran all verification tests (including new one) on Jet and Hera successfully. Also ran all tests by retrieving data from HPSS, rather than using staged data; all passed. Ran Fundamental tests on Jet as well.

- [x] hera.intel
  - Ran all verification tests (including new one) and also modifed all verification tests to pull obs data from HPSS (bypassing staged data). All passed.
- [x] cheyenne.intel
  - Ran MET_verification_only_vx, MET_ensemble_verification, and MET_ensemble_verification_only_vx
- [x] jet.intel
  - Ran all verification tests (including new one) and also modifed all verification tests to pull obs data from HPSS (bypassing staged data). Also ran fundamental tests.
- [x] fundamental test suite
  - Jet

**Note**: After rebasing latest changes, ran fundamental and all verification tests on Hera; all passed.

## DEPENDENCIES:
- ~~#853 should be merged prior to this PR; those changes are included in this PR~~

## DOCUMENTATION:
- Documentation added to Gillian's documentation updates here: https://github.com/gspetro-NOAA/ufs-srweather-app/pull/1

## ISSUE: 
 - Resolves #863
 - ~~Resolves #851~~ Fixed by #868
 - Resolves #850 
 - Resolves #808 

## CHECKLIST
- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

mkavulich committed:
Remove unnecessary "minus 1" dates

mkavulich committed:
Remove unused logic for 3- and 6-hour accumulation times

mkavulich committed:
Convert exregional_get_obs_ccpa.sh to use retrieve_data.py...this version may have bugs, I am committing so I can test on another platform since my Jet "fairshare" is apparently off-the-charts low

mkavulich committed:
New script working for CCPA obs 🥳

mkavulich committed:
Fix bug in retrieve_data.py: when the glob expression for files in the tarball contains a full path, we need to lstrip the leading "/" before globbing

mkavulich committed:
Consolidate all "get_obs" tasks for vx into single J-job/exscript.

All tasks now working for simple test case! Next up is to try all WE2E
vx tests

mkavulich committed:
Now that retrieve_data.py checks for the existence of the output directory, we need to create it prior to calling the script

mkavulich committed:
Complete transition to new obs pulling script. Currently works to pull
obs for tests MET_verification, MET_verification_only_vx, and
MET_ensemble_verification (modified to pull rather than use staged data)

 - Remove old j-jobs and exscripts
 - Cleanup and update mrms_pull_topofhour.py to be more pythonic
 - Move to obs dir prior to pulling data to avoid potential conflicts
from temporarily extracted files from HPSS
 - Move many comments from config_defaults.yaml to more appropriate
location in new data pulling/staging script; add more details about data
quirks

mkavulich committed:
Get rid of unneccessary directory structure requirement for obs. All verification tests now pass when retrieving from HPSS!

mkavulich committed:
Wildcards not allowed in 'archive_file_names' entry

mkavulich committed:
Updates to exregional_get_verif_obs.sh:

 - Remove usage of "*_vrfy" functions due to issues mentioned in Issue
   861 (https://github.com/ufs-community/ufs-srweather-app/issues/861)
 - Remove extra "mkdir" statement where it shouldn't be for MRMS
 - Use correct variable (vhh_noZero) for checking valid hour for NDAS

mkavulich committed:
Updates to verification tests:

 - Remove stanzas eliminating the "task_get_obs*" tasks; these should
   now be run every time (if using staged data, they will check that the
   data exists)
 - Modify tasks RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_RAP
   (MET_verification) and MET_ensemble_verification_only_vx_time_lag to
   retrieve obs data from HPSS
 - Fix MET_ensemble_verification_only_vx_time_lag test by correcting
   variable NET to NET_default (addresses issue 851)
 - Add new verification task custom_ESGgrid_Eastern_NA_snow_8km that
   will have winter weather verification tasks (for subsequent PR)

mkavulich committed:
Fix write component for new test

mkavulich committed:
Replace "#!/bin/bash" with "#!/usr/bin/env bash" per Issue 857

mkavulich committed:
Update some comments and prints for new j-Job

mkavulich committed:
Make script mrms_pull_topofhour.py raise an exception if no file is found

mkavulich committed:
Remove immediate quit when specified output path does not exist; this didn't account for templating behavior

mkavulich committed:
Fix error in output directory name for test "test_gdas_ics_from_aws"

mkavulich committed:
Roll new snowfall/NOHRSC verification into new data retrieval system

 - Remove stand-alone j-jobs and exscripts for NOHRSC
 - Add NOHRSC_obs stanza to parm/data_locations.yml
 - Update documentation and comments to include NOHRSC obs
 - Add "ASNOW" (accumulated snow) verification to
   custom_ESGgrid_Great_Lakes_snow_8km
 -

mkavulich committed:
More testing changes/updates

 - Don't retrieve obs data from HPSS for
   grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_RAP; too many
   platforms use this test
 - Rename custom_ESGgrid_Eastern_NA_snow_8km to
   custom_ESGgrid_Great_Lakes_snow_8km; add to comprehensive and Jet
   coverage test
 - Add MET_ensemble_verification_only_vx to comprehensive for Cheyenne
   and Orion, as well as to coverage suite via Hera/GNU
 - Move MET_verification_only_vx coverage from Hera/GNU to Cheyenne/GNU
 - Add new mysterious "2020_CAD" test to all comprehensive lists, not
   just Orion (what's up with that?), sort alphabetically

mkavulich commented:
@MichaelLueken This branch has been rebased with the latest changes from develop, so should be in its final state pending reviewer comments.

MichaelLueken commented on tests/WE2E/machine_suites/comprehensive.cheyenne:
```diff
@@ -1,3 +1,4 @@
+2020_CAD
```
I think that we can include the 2020_CAD test on all machines except for Cheyenne.  The staged data was placed in `/glade/work/epicufsrt/contrib/UFS_SRW_data/develop/input_model_data/FV3GFS/nemsio/2020020312`.  As the test config is currently set up, I think that it will attempt to look for the data in `/glade/p/ral/jntp/UFS_CAM/COMGFS/gfs.20200203/12` and the test will fail since the data isn't available.

mkavulich replied:
Ah I had forgotten that Cheyenne can't automatically get AWS data. I had just noticed that test was only added to the "Orion" comprehensive list when it shouldn't be restricted to just that machine. I have made that update.

Regarding the new `custom_ESGgrid_Great_Lakes_snow_8km` test and what platforms it runs on, the idea behind the "comprehensive" test suite is that *all* tests that can run on a given platform should be included. So unless a test is known to fail, it should go in the `comprehensive` file. If Gaea can't run HPSS tests, then it should have its own `comprehensive.gaea` file that omits tests requiring HPSS access. I had just assumed Gaea had HPSS access since it didn't have its own file yet (I don't have access to this machine).

MichaelLueken replied:
It makes sense to add the new `custom_ESGgrid_Great_Lakes_snow_8km` to the `comprehensive` test file.  Technically, the SRW App doesn't build or run on Gaea currently, so including this test in the `comprehensive` test file will work for now.  Once the SRW App is able to build and run on Gaea, I will try running the `comprehensive` test file on that machine and create a new `comprehensive.gaea` file that omits the tests requiring HPSS access from that suite if those tests fail.

MichaelLueken reviewed (commented):
@mkavulich -

Overall these changes look good to me!

I do have some concerns about adding the `2020_CAD` test to `comprehensive.cheyenne` and the new `custom_ESGgrid_Great_Lakes_snow_8km` to `comprehensive`.

The `2020_CAD` test only has data staged on Cheyenne.  All other machines are able to pull data from AWS, but that wasn't possible on Cheyenne.  Since staged data isn't available on all machines, there is no `USE_USER_STAGED_EXTRN_FILES` in the test's config file.  In order to add this test to `comprehensive.cheyenne`, the staged data on Cheyenne would need to be propagated to the rest of the machines and `USE_USER_STAGED_EXTRN_FILES: true` would need to be added.  It should be fine to add this test to `comprehensive`, but should be removed from `comprehensive.cheyenne` (unless the data is added to the rest of the Tier-1 platforms).

The new `custom_ESGgrid_Great_Lakes_snow_8km` test requires access to HPSS to pull the necessary verification data.  While this test has been successfully run on both Hera and Jet, Gaea doesn't have the necessary `hpss` module that can be loaded to pull the data from HPSS.  Would adding this test to both the `coverage.jet` and `coverage.hera.gnu.com` (or `coverage.hera.intel.nco`) suites and removing it from `comprehensive` be the better path forward?

mkavulich committed:
Cheyenne can't automatically get AWS data, removing 2020_CAD test from comprehensive.cheyenne

MichaelLueken reviewed (approved):
@mkavulich -

Thanks for addressing my concern regarding the `2020_CAD` test in the `comprehensive.cheyenne` file!  Approving these changes now.

VanderleiVargas-NOAA commented:
@mkavulich 

It seems that some of the tests are failing on Hera. The following tests succeeded:

grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_RAP
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16
MET_ensemble_verification_only_vx
MET_verification_only_vx

While the following tests failed:

custom_ESGgrid_Great_Lakes_snow_8km
MET_ensemble_verification_only_vx_time_lag

On the develop branch all tests succeeded except for custom_ESGgrid_Great_Lakes_snow_8km. Since this last one is a new config file, I simply copied it and tried to use it on the develop branch.

In all the cases where the run did not succeed, the code stopped while executing the following tasks: get_obs_ccpa,  get_obs_nohrsc, get_obs_mrms, get_obs_ndas.

mkavulich commented:
@VanderleiVargas-NOAA can you point me to your tests on Hera? I have run those tests on Hera (with Intel) and they passed on my end.

VanderleiVargas-NOAA commented:
@mkavulich 

the outputs are at 
/scratch2/BMC/fv3lam/Vanderlei.Vargas/expt_dirs/custom_grid_issue_850

the SRW app that I used is at
/scratch2/BMC/fv3lam/Vanderlei.Vargas/ufs-srweather-app_mike

mkavulich commented:
@VanderleiVargas-NOAA Those two tests pull data from HPSS and it looks like you don't have HPSS permissions yet. We'll need to get you added to the right permissions group, we can discuss it at this week's meeting.

VanderleiVargas-NOAA commented:
@mkavulich Thanks for taking a look at it. That's good to know!

mkavulich commented:
Thanks for taking the time to try it out! Let me know if you have any more comments on the code or the other tests.

michelleharrold reviewed (commented):
@mkavulich -- I looked over the PR, and it looks like it is in great shape.

gsketefian commented on jobs/JREGIONAL_GET_VERIF_OBS:
```diff
@@ -1,9 +1,9 @@
 #-----------------------------------------------------------------------
 #
-# This script pulls NDAS observation data for comparison to the model.
+# This script checks, pulls. and stages observation data for model verification.
```
pulls. -> pulls,

gsketefian commented on scripts/exregional_get_verif_obs.sh:
```diff
@@ -0,0 +1,456 @@
+
+  # Calculate valid date + 1 day; this is needed because (for some ungodly reason) CCPA files for 19-23z
+  # are stored in the *next* day's 00h directory
+  vdate_p1=`$DATE_UTIL -d "${unix_init_DATE} ${current_fcst} hours 1 day" +%Y%m%d%H`
```
@mkavulich Use more "modern" `$(...)` notation to enclose commands as opposed to backticks (here and elsewhere).

gsketefian commented on tests/WE2E/test_configs/verification/config.MET_ensemble_verification_only_vx_time_lag.yaml:
```diff
@@ -32,9 +29,9 @@ global:
-  CCPA_OBS_DIR: '/scratch2/BMC/fv3lam/ens_design_RRFS/obs_data/ccpa/proc'
-  MRMS_OBS_DIR: '/scratch2/BMC/fv3lam/ens_design_RRFS/obs_data/mrms/proc'
-  NDAS_OBS_DIR: '/scratch2/BMC/fv3lam/ens_design_RRFS/obs_data/ndas/proc'
+  CCPA_OBS_DIR: '{{ workflow.EXPTDIR }}/obs_data/ccpa/proc'
```
@mkavulich Can we make these the defaults in `config_defaults.yaml`?  Seems like this is the logical default place obs should be placed in.

mkavulich replied:
@gsketefian I have made this the default, and added some special functionality to run_WE2E_tests.py to overwrite with the machine file values in the test specifically. However, this still has to be specified here for supported platforms because otherwise the Machine File value would overwrite the default (users will *not* have to specify these variables, this is only for the testing script). This is kind of a chicken-egg problem if we want to both have staged data and the ability to automatically retrieve custom data.

gsketefian commented on ush/config_defaults.yaml:
```diff
@@ -246,111 +246,41 @@ platform:
+  # in file scripts/exregional_get_verif_obs.sh for more details about
+  # files and directory structure, as well as important caveats about
+  # errors in the metadata and file names.
+  # NOTE: This should not be set equal to other *OBS_DIR variables,
```
@mkavulich What variables are you referring to by `*OBS_DIR`?

gsketefian replied:
Oh, you mean don't set `CCPA_OBS_DIR` to say `MRMS_OBS_DIR`?  If so, can you clarify?  Thanks.

gsketefian commented on tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_RAP.yaml:
```diff
@@ -21,9 +21,6 @@ rocoto:
     metatask_run_ensemble:
       task_run_fcst_mem#mem#:
         walltime: 01:00:00
```
@mkavulich So are these lines being removed to enable the `get_obs_...` tasks?  If so, does that mean we don't need to stage the obs for these tests?

mkavulich replied:
I simply added a line to the script that checks if all the observations already exist at the specified location. If they already exist, we do nothing, otherwise we attempt to pull them from HPSS. This removes the need for de-activating the tasks depending on where you want to get your observations.

gsketefian commented:
@mkavulich Given that none of my comments are critical and I'm out the next 1.5 weeks (and since @michelleharrold already approved), I will approve this PR now.  It would be nice if you addressed my minor comments/questions before merging.  Thanks!  @MichaelLueken

MichaelLueken commented:
Adding DO_NOT_MERGE label until @mkavulich has addressed @gsketefian's comments/questions.

mkavulich committed:
Address Gerard's comments

mkavulich committed:
Move machine file definitions of *_OBS_DIRs to TEST_*_OBS_DIR for WE2E testing purposes, similar to TEST_ALT_EXTRN_MDL_SYSBASEDIR_ICS

mkavulich commented:
@gsketefian I think I have addressed all your comments. If I have missed any we can address them when you get back.

@MichaelLueken I don't think the DO_NOT_MERGE label is necessary since these are optional comments. I am also out next week, so I would appreciate if we could get the Jenkins tests run today to make sure they all pass.

MichaelLueken commented:
@mkavulich -

I've noted a failure in the following Hera GNU test:

`MET_ensemble_verification_only_vx_time_lag` failed due to permission issues attempting to open MRMS and NDAS HPSS files:

```
ERROR: [FATAL] no permission to open HPSS archive file: /NCEPPROD/hpssprod/runhistory/rh2021/202105/20210505/dcom_prod_ldmdata_obs.tar
HTAR: HTAR FAILED
```

Please see `/scratch1/NCEPDEV/stmp2/role.epic/jenkins/workspace/fs-srweather-app_pipeline_PR-864/expt_dirs/MET_ensemble_verification_only_vx_time_lag` for the Jenkins test log files.

Hera GNU tests are still running and Hera Intel pipeline has just queued, so will be awhile before those complete.

Orion is experiencing issues, so I will likely need to requeue those tests on Monday.

I've noted failures in the following tests on Jet:

`custom_ESGgrid_Great_Lakes_snow_8km` failed due to permission issues attempting to open MRMS, NDAS, and NOHRSC  HPSS files:

```
ERROR: [FATAL] no permission to open HPSS archive file: /NCEPPROD/hpssprod/runhistory/rh2023/202302/20230217/dcom_ldmdata_obs.tar
HTAR: HTAR FAILED
```
Please see `/mnt/lfs4/HFIP/hfv3gfs/role.epic/jenkins/workspace/fs-srweather-app_pipeline_PR-864/expt_dirs/custom_ESGgrid_Great_Lakes_snow_8km` for the Jenkins test log files.

Most of the Jet tests are still running currently, so more issues might arise as the tests continue to run.

The main issue right now appears to be that MRMS, NDAS, and NOHRSC HPSS archive files contain restricted data and the EPIC role account doesn't have rstprod permissions.  It looks like we'll need to exercise staged data for all verification tests.

mkavulich commented:
@MichaelLueken Given that retrieving data is a capability we want to test, is there not any way to request the proper permissions? The retrieved data will never leave the platforms they are tested on (Hera and Jet) so I don't see why that would be a problem.

mkavulich commented:
As a follow up: I'm not sure about this, but I *think* that we don't actually pull any restricted data, it's just all stored in the same place which is why permissions are a problem. I'd have to check with someone who knows more about the obs, but I have to imagine that would be another good justification for getting the permissions. 

As I mentioned at the meeting, I will be on PTO for the next week, but I will check in as soon as I can when I get back to answer any follow-up questions.

MichaelLueken commented:
@mkavulich -

In the past, EPIC has reached out and asked the sys admins on NOAA RDHPCS about the process for adding rstprod permission to the EPIC role account.  Everyone with access to the role account will need rstprod before access can be granted to the role account.  So, it will take time to get everyone onboard with rstprod permissions.  Just to note, we have had epic project on all RDHPCS platforms for several months, but the role account still hasn't been updated to use these allocations (apparently due to users of the EPIC role account not requesting access to the epic project on AIM and RDHPCS not treating this a priority).

If it is possible to pull just the verification data from HPSS and not the restricted data, that would likely be the best and fastest approach.  If that isn't a possibility, then we will need to fall back to using staged data until the role account and everyone using it has rstprod permissions.

MichaelLueken commented:
@mkavulich -

The official request for the EPIC role account to acquire rstprod access has been sent to the RDHPCS AIM helpdesk.  So, in the future, this should no longer be an issue.  However, it is likely going to take a bit for rstprod access to be granted to the role account (everyone on the EPIC role account will need to acquire rstprod access before the role account can gain access).  How would you like to proceed with getting this work merged?  If it is possible to pull the CCPA, MRMS, NOHRSC, and NDAS data without the restricted data included, that would work.  Alternatively, I can include these tests as those that need to be manually run (similar to Cheyenne tests) before merging.

mkavulich commented:
@MichaelLueken Thanks for moving that along! I just got back from PTO so I'm a little behind, but I plan on bringing this PR up to date to eliminate the merge conflicts today. 

Given that the ball is rolling I think it's acceptable to temporarily omit these tests from the coverage tests and run them manually (so long as you're okay with that). I will comment these tests out from the coverage list until the role account can get the necessary permissions.

MichaelLueken commented:
@mkavulich - I think that temporarily commenting out the tests in question and manually running them until the role account has the necessary permissions to run them is fine.

mkavulich committed:
Temporarily remove HPSS-pulling verification tests from coverage suites

mkavulich commented:
@MichaelLueken The conflicts have been resolved and I commented out the two tests we discussed from the coverage suite files. Let me know if you need anything else from my end.

Edit: to ensure the conflict resolution didn't break anything major I ran the verification and fundamental tests on Hera/Intel

MichaelLueken commented:
The Cheyenne Intel WE2E coverage tests were run on Hera and all successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
custom_GFDLgrid__GFDLgrid_USE_NUM_CELLS_IN_FILENAMES_eq_FALSE      COMPLETE               7.24
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot     COMPLETE              29.03
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16                COMPLETE              19.14
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_HRRR           COMPLETE              25.14
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE               8.43
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_HRRR_suite_HRRR                COMPLETE              16.38
pregen_grid_orog_sfc_climo                                         COMPLETE               6.25
specify_template_filenames                                         COMPLETE               7.25
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             118.86
```
The Cheyenne GNU WE2E coverage tests were run on Hera and all successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
grid_CONUS_25km_GFDLgrid_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16      COMPLETE              20.57
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta      COMPLETE             228.63
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              28.11
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR             COMPLETE              37.18
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP              COMPLETE              82.52
grid_RRFS_CONUScompact_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16   COMPLETE              22.53
grid_RRFS_NA_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP                 COMPLETE             314.56
grid_SUBCONUS_Ind_3km_ics_NAM_lbcs_NAM_suite_GFS_v16               COMPLETE              49.54
MET_verification_only_vx                                           COMPLETE               0.12
specify_EXTRN_MDL_SYSBASEDIR_ICS_LBCS                              COMPLETE              10.07
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             793.83
```
Manual testing of MET_ensemble_verification_only_vx_time_lag on Hera GNU:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
MET_ensemble_verification_only_vx_time_lag                         COMPLETE               7.67
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE               7.67
```
The Gaea WE2E coverage tests were manually run and all successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
community                                                          COMPLETE              20.68
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE              26.89
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP              COMPLETE              30.83
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR             COMPLETE              33.32
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson  COMPLETE             354.19
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR          COMPLETE              31.69
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta     COMPLETE             355.89
grid_SUBCONUS_Ind_3km_ics_RAP_lbcs_RAP_suite_RRFS_v1beta_plot      COMPLETE               9.69
nco_ensemble                                                       COMPLETE              81.59
nco_grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thom  COMPLETE             349.73
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            1294.50
```
The Jet WE2E coverage tests were manually run and all successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
community                                                          COMPLETE              17.01
custom_ESGgrid                                                     COMPLETE              17.80
custom_GFDLgrid                                                    COMPLETE               8.46
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2021032018         COMPLETE               9.85
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_netcdf_2022060112_48h     COMPLETE              52.35
get_from_HPSS_ics_RAP_lbcs_RAP                                     COMPLETE              16.02
grid_RRFS_AK_3km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR                 COMPLETE             235.16
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot     COMPLETE              32.01
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2        COMPLETE               7.20
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta       COMPLETE             523.06
nco_grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR       COMPLETE              10.77
process_obs                                                        COMPLETE               0.22
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             929.91
```
Manual testing of custom_ESGgrid_Great_Lakes_snow_8km on Jet:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
custom_ESGgrid_Great_Lakes_snow_8km                                COMPLETE              15.17
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE              15.17
```
The Hera and Orion Jenkins tests successfully passed.  Now moving forward with merging this work.
